Formula/h/hdf5.rb

```diff
@@ -43,32 +42,18 @@ def install
               "settingsdir=$(libdir)",
               "settingsdir=#{pkgshare}"
 
-    system "autoreconf", "--force", "--install", "--verbose"
-
-    args = %W[
-      --disable-dependency-tracking
-      --disable-silent-rules
-      --enable-build-mode=production
-      --enable-fortran
-      --enable-cxx
-      --prefix=#{prefix}
-      --with-szlib=#{Formula["libaec"].opt_prefix}
-    ]
-    args << "--with-zlib=#{Formula["zlib"].opt_prefix}" if OS.linux?
-
-    system "./configure", *args
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
```

Comment:
try to add "-DCMAKE_CXX_FLAGS=-fmacro-prefix-map=$PWD=.", "-DCMAKE_C_FLAGS="-fmacro-prefix-map=$PWD=." or something similar to replace absolute paths in error messages with a dot.